rust/osm2lanes/src/overpass.rs

```diff
@@ -41,3 +46,69 @@ pub async fn get_way(id: ElementId) -> Result<Tags, reqwest::Error> {
     assert_eq!(element.id, id);
     Ok(element.tags)
 }
+
+pub async fn get_iso3166_2(id: ElementId) -> Result<String, reqwest::Error> {
+    let mut resp = reqwest::Client::new()
+        .get(format!(
+            r#"https://overpass-api.de/api/interpreter?data=[out:json][timeout:2];way(id:{});>;is_in;area._["ISO3166-2"];out tags;"#,
+            id
+        ))
+        .send()
+        .await?
+        .json::<OverpassResult>()
+        .await?;
+    log::debug!("{:#?}", resp);
+    assert_eq!(resp.elements.len(), 1);
+    let element = resp.elements.pop().unwrap();
+    Ok(element.tags.get("ISO3166-2").unwrap().to_owned())
+}
+
+pub async fn get_way(id: ElementId) -> Result<(Tags, Locale), reqwest::Error> {
+    let resp = reqwest::Client::new()
+        .get(format!(
+            r#"https://overpass-api.de/api/interpreter?data=[out:json][timeout:25];
+            way(id:{id});
+            out tags;
+            way(id:{id});
+            >;
+            is_in->.enclosing;
+            (
+              area.enclosing["ISO3166-2"];
+              area.enclosing["ISO3166-1"];
+              area.enclosing["driving_side"];
+            );
+            out tags;"#
+        ))
+        .send()
+        .await?
+        .json::<OverpassResult>()
+        .await?;
+    log::debug!("{:#?}", resp);
+
+    let iso3166_2 = resp
+        .elements
+        .iter()
+        .find_map(|element| element.tags.get("ISO3166-2"));
+    let iso3166_1 = resp
+        .elements
+        .iter()
+        .find_map(|element| element.tags.get("ISO3166-1"));
+    let driving_side = resp
+        .elements
+        .iter()
+        .find_map(|element| element.tags.get("driving_side"));
+    let locale = Locale::builder()
+        .driving_side(driving_side.map_or(DrivingSide::Right, |d| d.parse().unwrap()))
+        .iso_3166_option(iso3166_2.or(iso3166_1))
+        .build();
+
+    let way_element = {
+        let mut elements = resp.elements;
+        elements.truncate(1);
+        elements.pop().unwrap()
+    };
```

Comment:
suggestions on how to better consume a vector and just return the first item are welcome...

Reply:
Hmm. Performance is unlikely to matter here since we're not calling it in a tight loop (and networking lag dominates by far); why not just `resp.elements.remove(0)` and pay the cost of shifting the vector down? Alternatively, `swap_remove` is a good idea from https://www.reddit.com/r/learnrust/comments/hwseol/whats_the_idiomatic_way_to_move_the_first_element/

Reply:
yeah, so nothing significantly better than what I did for readability. I will leave it as is, thanks